code/__DEFINES/traits.dm

```diff
@@ -158,6 +158,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_XENO_IMMUNE		"xeno_immune"//prevents xeno huggies implanting skeletons
 #define TRAIT_NAIVE				"naive"
 #define TRAIT_GUNFLIP			"gunflip"
+#define TRAIT_SPACEWALK			"spacewalk"
```

Comment:
I'd argue making an inherent traits list on the mob is better because it's somewhat easier to set-up.

Reply:
it'd mean a copy of that list would exist for every copy of that mob.

Reply:
A trait list is also annoying to deal with when considering inheritance. Look at the golem species for an example: if you want one extra trait, you need to paste the entire list, and it's not very clear what was changed from the original if the list is long.

That said, using Initialize to set it up still causes complications if you want a subtype _without_ one of the default traits, so perhaps the ideal solution would be some other third option.

Reply:
A TraitInitialize? That way it can be inherited to add a few more traits, or just completely overridden if you want less. Still not perfect because you'd need to rewrite the whole thing if you intend to remove a single trait, but would probably be a decent third option.

Reply:
Maybe a TraitInitialize that returns a list of traits which are then added in the Initialize proper, so you can call ..(), subtract traits from the list, and return the result. More processing-heavy, though.

Reply:
@XDTM if a subtype shouldnt have a trait it shouldnt be a subtype, thats why traits exist.

Reply:
I don't see the logic in that.

I faced that issue with wood golems, i wanted them to have all the golem traits except being fireproof, so i ended up having to copy the full list minus the fireproof trait. Not making it a subtype would imply copypasting everything else the golem species does for no real reason.

Reply:
no it would imply you need to make the stuff golem species does into traits also

Reply:
how would you make golem name generation into a trait?

```
/datum/species/golem/random_name(gender,unique,lastname)
	var/golem_surname = pick(GLOB.golem_names)
	// 3% chance that our golem has a human surname, because
	// cultural contamination
	if(prob(human_surname_chance))
		golem_surname = pick(GLOB.last_names)
	else if(special_names && special_names.len && prob(special_name_chance))
		golem_surname = pick(special_names)

	var/golem_name = "[prefix] [golem_surname]"
	return golem_name
```

Reply:
Add this part of code to 
`datum/species/random_name()`
After 
`if(HAS_TRAIT(src, TRAIT_MINERAL_NAME))`

Reply:
that's terrible code though, and doesn't let golems add their own special custom name pools like they do now

this whole discussion is pointless because traits are just fancy flags and flags are just condensed boolean variables, so it's like saying that a base object with `var/thing = FALSE` can have a child with `thing = TRUE` but not viceversa for some arbitrary reason